# Nathan Fernandez - IS 362 - Project 2

## Dataset 1: Monthly Product Sales

This section analyzes a fictional dataset I created that shows monthly sales for three products at an online store.

The original dataset is organized in a wide format, with each month stored as a separate column. This structure is easy to read, but it is not ideal for some types of data analysis.

The goal of this section is to use Python and pandas to tidy the dataset by converting the monthly columns into a single Month variable and a Sales variable. The transformed data can then be used to analyze sales trends and compare the products over time.

In [1]:
import pandas as pd

sales = pd.read_csv("Project 2 Dataset 1.csv")

sales

,Product,January,February,March,April,May
0,Laptop Stand,42,51,47,63,58
1,Wireless Mouse,75,81,69,92,88
2,Keyboard,54,61,57,73,70


## Preparing the Data

The original dataset is organized in a wide format. Each product is represented by one row, while January through May are separate columns.

Although this format is easy to read, it is not ideal for analyzing sales trends across time. To make the dataset tidier, the month columns will be converted into values in a single `Month` column, and the corresponding sales amounts will be stored in a `Sales` column.

In [2]:
tidy_sales = sales.melt(
    id_vars="Product",
    var_name="Month",
    value_name="Sales"
)

tidy_sales

,Product,Month,Sales
0,Laptop Stand,January,42
1,Wireless Mouse,January,75
2,Keyboard,January,54
3,Laptop Stand,February,51
4,Wireless Mouse,February,81
5,Keyboard,February,61
6,Laptop Stand,March,47
7,Wireless Mouse,March,69
8,Keyboard,March,57
9,Laptop Stand,April,63


### Tidy Data Transformation

The `melt()` function was used to transform the dataset from wide format into tidy format. The five month columns were converted into values in the `Month` column, while their corresponding sales values were placed in the `Sales` column.

The resulting dataset contains 15 observations, representing five months of sales for each of the three products. Each row now represents one product's sales for one month, which makes the data easier to analyze by product and over time.

In [3]:
product_totals = tidy_sales.groupby("Product")["Sales"].sum()

product_totals

Product
Keyboard          315
Laptop Stand      261
Wireless Mouse    405
Name: Sales, dtype: int64

### Total Sales by Product

The total sales were calculated by grouping the tidy data by product and summing the sales values for each product.

The Wireless Mouse had the highest total sales with **405 units**. The Keyboard had **315 units**, while the Laptop Stand had **261 units**.

Based on these five months of data, the Wireless Mouse was the best-selling product overall.

In [4]:
monthly_totals = tidy_sales.groupby("Month")["Sales"].sum()

monthly_totals

Month
April       228
February    193
January     171
March       173
May         216
Name: Sales, dtype: int64

### Monthly Sales Trends

The monthly totals were calculated by grouping the tidy data by month and summing the sales for all three products.

April had the highest total sales with **228 units**, followed by May with **216 units**. January had the lowest total sales with **171 units**.

Overall, sales increased from January through February, decreased slightly in March, increased substantially in April, and then decreased slightly in May. This shows some variation in monthly sales, although the five-month dataset is too small to establish a longer-term seasonal pattern.

In [5]:
highest_sale = tidy_sales.loc[tidy_sales["Sales"].idxmax()]

highest_sale

Product    Wireless Mouse
Month               April
Sales                  92
Name: 10, dtype: object

### Highest Individual Monthly Sale

The highest individual monthly sale was identified using the `idxmax()` function, which locates the row containing the maximum sales value.

The Wireless Mouse had the highest individual monthly sales figure, with **92 units sold in April**. This was the highest sales value for any product in a single month in the dataset.

In [6]:
tidy_sales.info()

<class 'pandas.DataFrame'>
RangeIndex: 15 entries, 0 to 14
Data columns (total 3 columns):
 #   Column   Non-Null Count  Dtype
---  ------   --------------  -----
 0   Product  15 non-null     str  
 1   Month    15 non-null     str  
 2   Sales    15 non-null     int64
dtypes: int64(1), str(2)
memory usage: 492.0 bytes


## Conclusion for Dataset 1

The original sales dataset was transformed from a wide format into a tidy format using pandas. The `melt()` function converted the month columns into a single `Month` variable and placed the corresponding sales amounts into a `Sales` variable.

The analysis showed that the **Wireless Mouse** had the highest total sales across the five months, with **405 units**. April had the highest overall monthly sales with **228 units** across all three products. The highest individual monthly sales figure was **92 units**, recorded for the Wireless Mouse in April.

The tidy structure makes it easier to analyze sales by product and month and provides a more useful format for further analysis.

## Dataset 2: Digital Product Sales

This section analyzes a fictional dataset created by classmate Bria King for the Week 5 discussion. The dataset shows the number of units sold for four types of digital products over three months.

The original dataset is organized in a wide format, with each month stored as a separate column. This structure is easy to read, but it is not ideal for analyzing sales trends across time.

The goal of this section is to use Python and pandas to tidy the dataset by converting the monthly columns into a single `Month` variable and a `Units Sold` variable. The transformed data can then be used to compare product sales and examine changes over time.


In [7]:
import pandas as pd

digital_sales = pd.read_csv("Project 2 Dataset 2.csv")

digital_sales

,Product,January,February,March
0,Printable Wall Art,25,32,41
1,Digital Planners,40,28,35
2,Budget Trackers,18,24,30
3,Coloring Pages,12,19,27


## Preparing the Data

The original dataset is organized in a wide format. Each product is represented by one row, while January, February, and March are separate columns.

Although this format is easy to read, it is not ideal for analyzing sales trends across time. To make the dataset tidier, the month columns will be converted into values in a single `Month` column, and the corresponding number of units sold will be stored in a `Units Sold` column.

In [8]:
tidy_digital_sales = digital_sales.melt(
    id_vars="Product",
    var_name="Month",
    value_name="Units Sold"
)

tidy_digital_sales

,Product,Month,Units Sold
0,Printable Wall Art,January,25
1,Digital Planners,January,40
2,Budget Trackers,January,18
3,Coloring Pages,January,12
4,Printable Wall Art,February,32
5,Digital Planners,February,28
6,Budget Trackers,February,24
7,Coloring Pages,February,19
8,Printable Wall Art,March,41
9,Digital Planners,March,35


### Tidy Data Transformation

The `melt()` function was used to transform the dataset from wide format into tidy format. The three month columns were converted into values in the `Month` column, while their corresponding unit totals were placed in the `Units Sold` column.

The resulting dataset contains 12 observations, representing three months of sales for each of the four digital products. Each row now represents one product's sales for one month, which makes the data easier to analyze by product and over time.

In [9]:
product_totals = tidy_digital_sales.groupby("Product")["Units Sold"].sum()

product_totals

Product
Budget Trackers        72
Coloring Pages         58
Digital Planners      103
Printable Wall Art     98
Name: Units Sold, dtype: int64

### Total Sales by Product

The total units sold were calculated by grouping the tidy data by product and summing the `Units Sold` values for each product.

The **Digital Planners** had the highest total sales with **103 units**. The **Printable Wall Art** had **98 units**, followed by **Budget Trackers** with **72 units** and **Coloring Pages** with **58 units**.

Based on these three months of data, the Digital Planners were the best-selling product overall.

In [10]:
monthly_totals = tidy_digital_sales.groupby("Month")["Units Sold"].sum()

monthly_totals

Month
February    103
January      95
March       133
Name: Units Sold, dtype: int64

### Monthly Sales Trends

The monthly totals were calculated by grouping the tidy data by month and summing the `Units Sold` values for all four products.

**March** had the highest total sales with **133 units**, followed by **February** with **103 units**. **January** had the lowest total sales with **95 units**.

Overall, sales increased each month from January through March. The increase from January to March suggests that the digital products experienced positive sales growth during this three-month period.

In [11]:
highest_sale = tidy_digital_sales.loc[tidy_digital_sales["Units Sold"].idxmax()]

highest_sale

Product       Printable Wall Art
Month                      March
Units Sold                    41
Name: 8, dtype: object

### Highest Individual Monthly Sale

The highest individual monthly sale was identified using the `idxmax()` function, which locates the row containing the maximum value in the `Units Sold` column.

**Printable Wall Art** had the highest individual monthly sales figure, with **41 units sold in March**. This was the highest sales value for any product in a single month in the dataset.

In [12]:
tidy_digital_sales.info()

<class 'pandas.DataFrame'>
RangeIndex: 12 entries, 0 to 11
Data columns (total 3 columns):
 #   Column      Non-Null Count  Dtype
---  ------      --------------  -----
 0   Product     12 non-null     str  
 1   Month       12 non-null     str  
 2   Units Sold  12 non-null     int64
dtypes: int64(1), str(2)
memory usage: 420.0 bytes


## Conclusion for Dataset 2

The original digital product sales dataset was transformed from a wide format into a tidy format using pandas. The `melt()` function converted the month columns into a single `Month` variable and placed the corresponding sales amounts into a `Units Sold` variable.

The analysis showed that **Digital Planners** had the highest total sales across the three months, with **103 units**. March had the highest overall monthly sales with **133 units** across all four products. The highest individual monthly sales figure was **41 units**, recorded for Printable Wall Art in March.

The tidy structure makes it easier to analyze sales by product and month and provides a more useful format for further analysis.

## Dataset 3: NBA Player Statistics

This section analyzes a dataset created by classmate Elian Martinez for the Week 5 discussion. The dataset contains game statistics for four NBA players, including their team, points, rebounds, assists, and field goals made and attempted.

The original dataset is untidy because multiple variables are stored together in the `Game Stats` and `Shooting` columns. For example, the `Game Stats` column contains points, rebounds, and assists in a single field, while the `Shooting` column contains field goals made and attempted in a single field.

The goal of this section is to use Python and pandas to separate these combined values into individual variables. The transformed data can then be used to compare player performance and shooting statistics.

In [13]:
nba_stats = pd.read_csv("Project 2 Dataset 3.csv")

nba_stats

,Player,Team,Game Stats,Shooting
0,Marcus Reed,New York,"24 PTS, 7 REB, 5 AST",9/17 FG
1,Jordan Ellis,Miami,"18 PTS, 11 REB, 3 AST",7/14 FG
2,Devin Carter,Boston,"29 PTS, 4 REB, 8 AST",11/20 FG
3,Andre Lewis,Chicago,"21 PTS, 6 REB, 7 AST",8/16 FG


## Preparing the Data

The original dataset contains multiple statistics combined within individual columns. The `Game Stats` column stores points, rebounds, and assists together, while the `Shooting` column stores field goals made and field goals attempted together.

To create a tidier dataset, these combined values will be separated into individual columns for `Points`, `Rebounds`, `Assists`, `FG Made`, and `FG Attempted`. This will make each statistic its own variable and make the data easier to analyze and compare across players.

In [14]:
nba_tidy = nba_stats.copy()

nba_tidy[["Points", "Rebounds", "Assists"]] = (
    nba_tidy["Game Stats"]
    .str.extract(r"(\d+)\s*PTS,\s*(\d+)\s*REB,\s*(\d+)\s*AST")
    .astype(int)
)

nba_tidy

,Player,Team,Game Stats,Shooting,Points,Rebounds,Assists
0,Marcus Reed,New York,"24 PTS, 7 REB, 5 AST",9/17 FG,24,7,5
1,Jordan Ellis,Miami,"18 PTS, 11 REB, 3 AST",7/14 FG,18,11,3
2,Devin Carter,Boston,"29 PTS, 4 REB, 8 AST",11/20 FG,29,4,8
3,Andre Lewis,Chicago,"21 PTS, 6 REB, 7 AST",8/16 FG,21,6,7


In [15]:
nba_tidy[["FG Made", "FG Attempted"]] = (
    nba_tidy["Shooting"]
    .str.extract(r"(\d+)/(\d+)\s*FG")
    .astype(int)
)

nba_tidy

,Player,Team,Game Stats,Shooting,Points,Rebounds,Assists,FG Made,FG Attempted
0,Marcus Reed,New York,"24 PTS, 7 REB, 5 AST",9/17 FG,24,7,5,9,17
1,Jordan Ellis,Miami,"18 PTS, 11 REB, 3 AST",7/14 FG,18,11,3,7,14
2,Devin Carter,Boston,"29 PTS, 4 REB, 8 AST",11/20 FG,29,4,8,11,20
3,Andre Lewis,Chicago,"21 PTS, 6 REB, 7 AST",8/16 FG,21,6,7,8,16


In [16]:
nba_tidy = nba_tidy.drop(columns=["Game Stats", "Shooting"])

nba_tidy

,Player,Team,Points,Rebounds,Assists,FG Made,FG Attempted
0,Marcus Reed,New York,24,7,5,9,17
1,Jordan Ellis,Miami,18,11,3,7,14
2,Devin Carter,Boston,29,4,8,11,20
3,Andre Lewis,Chicago,21,6,7,8,16


### Tidy Data Transformation

The original dataset stored multiple statistics together in the `Game Stats` and `Shooting` columns. These combined values were separated into individual columns for `Points`, `Rebounds`, `Assists`, `FG Made`, and `FG Attempted`.

After extracting the individual statistics, the original `Game Stats` and `Shooting` columns were removed because their information was now represented by the new variables. The resulting dataset has one column for each variable, making it easier to compare and analyze the players' performance.

In [17]:
nba_tidy["FG Percentage"] = (
    nba_tidy["FG Made"] / nba_tidy["FG Attempted"] * 100
).round(2)

nba_tidy

,Player,Team,Points,Rebounds,Assists,FG Made,FG Attempted,FG Percentage
0,Marcus Reed,New York,24,7,5,9,17,52.94
1,Jordan Ellis,Miami,18,11,3,7,14,50.00
2,Devin Carter,Boston,29,4,8,11,20,55.00
3,Andre Lewis,Chicago,21,6,7,8,16,50.00


### Field Goal Percentage

Field goal percentage was calculated by dividing each player's field goals made by their field goals attempted and multiplying by 100.

Devin Carter had the highest field goal percentage at **55.00%**, followed by Marcus Reed at **52.94%**. Jordan Ellis and Andre Lewis both had field goal percentages of **50.00%**.

In [18]:
stat_leaders = nba_tidy.loc[
    [
        nba_tidy["Points"].idxmax(),
        nba_tidy["Rebounds"].idxmax(),
        nba_tidy["Assists"].idxmax()
    ],
    ["Player", "Points", "Rebounds", "Assists"]
]

stat_leaders

,Player,Points,Rebounds,Assists
2,Devin Carter,29,4,8
1,Jordan Ellis,18,11,3
2,Devin Carter,29,4,8


### Player Performance

The player statistics show that Devin Carter led the group in two categories. He scored the most points with **29** and also recorded the most assists with **8**.

Jordan Ellis led the group in rebounds with **11**. These results show how the tidy dataset makes it easier to identify and compare the leaders in individual statistical categories.

In [19]:
nba_tidy.info()

<class 'pandas.DataFrame'>
RangeIndex: 4 entries, 0 to 3
Data columns (total 8 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Player         4 non-null      str    
 1   Team           4 non-null      str    
 2   Points         4 non-null      int64  
 3   Rebounds       4 non-null      int64  
 4   Assists        4 non-null      int64  
 5   FG Made        4 non-null      int64  
 6   FG Attempted   4 non-null      int64  
 7   FG Percentage  4 non-null      float64
dtypes: float64(1), int64(5), str(2)
memory usage: 388.0 bytes


## Conclusion for Dataset 3

The original NBA player statistics dataset was transformed into a tidier format using pandas. The combined values in the `Game Stats` column were separated into individual variables for points, rebounds, and assists, while the `Shooting` column was separated into field goals made and field goals attempted. The original combined columns were then removed.

The analysis showed that **Devin Carter** led the group in points with **29** and assists with **8**. **Jordan Ellis** recorded the most rebounds with **11**. After calculating field goal percentage, Devin Carter also had the highest shooting percentage at **55.00%**.

The final dataset contains no missing values and stores each player statistic as its own variable. This tidy structure makes the data easier to compare and provides a more useful format for further analysis.